<!-- Mission de suivi de waypoints -->
<!-- Les waypoints sont envoyes par RVIZ -->
<launch>
	<!-- ACTUATORS -->
	<node name="motors" pkg="actuators" type="motor_controller.py">
		<param name="max_speed" value="6800"/>
		<param name="min_speed" value="5800"/>
		<param name="max_rot" value="7500"/>
		<param name="min_rot" value="4500"/>
		<!-- The motor takes without ramp  -->
		<remap from="cmd_vel_ramped" to="cmd_vel"/>
	</node>



	<!-- SENSORS -->
	<!-- IMU -->
	<node name="imu_boat" pkg="razor_imu_9dof" type="imu_node.py">
      <remap from="imu" to="imu_boat" />
      <rosparam file="$(find sensors)/config/imu_boat.yaml" command="load"/>
    </node>

    <!-- GPS -->
    <node name="gps_driver" pkg="nmea_navsat_driver" type="nmea_serial_driver">
		<param name="port" value="/dev/serial/by-id/usb-Prolific_Technology_Inc._USB-Serial_Controller_D-if00-port0"/>
    	<param name="baud" value="4800"/>
	</node>




	
	<!-- PROCESS -->
	<node name="regulateur_champs" pkg="process" type="regulateur_champs_boat2.py" output="screen">
		<param name="V0" value="6000"/>
		<param name="thetadot0" value="6000"/>
		<param name="vHigh" value="6700"/>
		<param name="vLow" value="6400"/>
		<param name="Kp" value="2000"/>
		<param name="Kd" value="0"/>
		<param name="V_lim_reverse" value="1.5"/>
		<param name="reverse_motor" value="-1"/> 	<!-- (-1) to reverse -->
	</node>

	<node name="behavior_server" pkg="process" type="behavior_server.py">
		<param name="show_plot" value="False"/>
	</node>


	<!-- UTILITIES -->
	<node name="point_clicker" pkg="utilities" type="clicked_to_waypoint.py"/>

	<node name="local_pose_convertor" pkg="utilities" type="LL_to_local.py">
		<remap from="imu" to="imu_boat"/>
	</node>



	<!-- AFFICHAGE -->
	<node pkg="rviz_satellite" type="map2world_tf_broadcaster.py"
        name="broadcaster_fixed" />
</launch>
